### ISSUE TITLE
Is there a syntax for (external) self-transitions?

### ISSUE THREAD

**Issue by Ulenspiegel** [boost-experimental/msm-lite#54]
I'm trying this snippet:

```c++
#include <cstdio>
#include <boost/msm-lite.hpp>

namespace msm = boost::msm::lite;

using msm::state;
using msm::event;
using msm::sm;
using msm::make_transition_table;

struct e1 {};

state<class S> s;

struct SM {
  auto configure() const noexcept {
    using namespace msm;
    using msm::on_exit;
    state<class Init> init;
    return make_transition_table(
        *init = s,
        s + event<e1> / [] { printf("Action: S -> S\n"); } = s,
        s + on_entry / [] { printf("Entering S.\n"); },
        s + on_exit / [] { printf("Leaving S.\n"); }
    );
  }
};

int main(int argc, char** argv) {
  sm<SM> m;
  printf("Processing E1\n");
  m.process_event(e1{});
  return 0;
}
```

It prints:

```
Entering S.
Processing E1
Action: S -> S
```

So, during `s -> s` transition the action is executed, but it appears that transition doesn't actually happen, so entry/exit actions aren't called again (logging shows no transition, either). Is there a way to make a self-transition?

**Comment by krzysztof-jusiak** [comment]
Thanks @Ulenspiegel, this is a bug, it should trigger entry/exit in that case, however, msm-lite currently treats the following cases the same way...

1. s + event // internal-transition, no on_entry/on_exit
2. s + event = s // self-transition, should trigger on_entry/on_exit

Long story short, your code should work as you expected, I will fix it ASAP.

**Comment by krzysztof-jusiak** [comment]
Fixed by 4058e776a781ed2167814083e29b3fd98d65338b. Syntax is the same as used in your example.
